app/controllers/product_categories_controller.rb

```diff
@@ -0,0 +1,22 @@
+class ProductCategoriesController < ApplicationController
+  def create
+    @product = Product.find(params[:product_id])
+    ProductCategory.create(product_category_params)
+    redirect_to @product
+  end
+
+  def destroy
+    @product = Product.find(params[:product_id])
```

Comment:
This is a nitpick, but since you have to first destroy the `@product_category` object, you shouldn't retrieve the `@product` object until after the `destroy` operation is completed, in case there is an issue with that, you would not have retrieved an unnecessary object.